server/src/main/java/org/spine3/server/entity/AbstractEntity.java

```diff
@@ -157,12 +160,126 @@ private static IllegalStateException noSuchConstructor(String entityClass, Strin
     }
 
     /**
+     * Obtains constructor for the passed aggregate part class.
+     *
+     * <p>The part class must have a constructor with ID and {@code AggregateRoot} parameters.
+     *
+     * @param entityClass the {@code AggregatePart} class
+     * @param idClass     the class of entity identifiers
+     * @param <E>         the entity type
+     * @param <I>         the ID type
+     * @return the {@code AggregatePart} constructor
+     * @throws IllegalStateException if the entity class does not have the required constructor
+     */
+    public static <E extends AggregatePart, I, R extends AggregateRoot<I>> Constructor<E>
+    getAggregatePartConstructor(Class<E> entityClass, Class<R> rootClass, Class<I> idClass) {
+        final Constructor<E> result = getAggregatePartSupertypeCtr(entityClass, idClass);
+        if (result != null) {
+            return result;
+        }
+
+        try {
+            final Constructor<E> constructor =
+                    entityClass.getDeclaredConstructor(idClass, rootClass);
+            constructor.setAccessible(true);
+            return constructor;
+        } catch (NoSuchMethodException e) {
+            final String errMsg = format("%s class must declare a constructor " +
+                                         "with ID and AggregateRoot parameters.", entityClass);
+            throw new IllegalStateException(errMsg, e);
+        }
+    }
+
+    /**
+     * Obtains the constructor for the passed aggregate part class.
+     *
+     * <p>Constructor will be returned if the first argument is aggregate ID
+     * and the second is {@code AggregateRoot}. For example:
+     * AggregatePartConstr(AnAggregateId id, AggregateRoot root).
+     *
+     * <p>Constructor won't be returned if the second constructor parameter
+     * is subtype of the {@code AggregateRoot}. For example:
+     * SubAggregateRoot extends AggregateRoot{...};
+     * AggregatePartConstr(AnAggregateId id, SubAggregateRoot root).
+     *
+     * @param entityClass the {@code AggregatePart} class
+     * @param idClass     the ID class
+     * @param <E>         the entity type
+     * @param <I>         the ID type
+     * @return the obtained constructor, otherwise {@code null}
+     */
+    @Nullable
+    @SuppressWarnings("unchecked")
+    // It is OK because the constructor arguments are checked, before returning the constructor.
+    private static <E extends AggregatePart, I> Constructor<E> getAggregatePartSupertypeCtr
```

Comment:
These `static` reflection-based methods seem to be a part of an external utility class. Please consider that to make the `AbstractEntity` code cleaner.